cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeClient.java

```diff
@@ -122,14 +127,29 @@ public void sendMessages(ProgramRunId programRunId,
       try (Reader reader = new InputStreamReader(urlConn.getInputStream(), StandardCharsets.UTF_8)) {
         ProgramRunInfo responseBody = GSON.fromJson(reader, ProgramRunInfo.class);
         if (responseBody.getProgramRunStatus() == ProgramRunStatus.STOPPING) {
-          // TODO - CDAP-18744 - terminate the program based on the program endtimestamp payload in response body
+          // this will trigger controller.stop() in DefaultRuntimeJob
```

Comment:
Remove this comment. What get triggered after the future completed is beyond this class knowledge.